### PR TITLE
fix: fix esm build & add vitest

### PR DESCRIPTION
**Pull request by hwbllmnn**
Fixes the esm build and gets rid of jest in favour of vitest.

@geostyler/devs Please review.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,15 +1,11 @@
 {
   "name": "geostyler-cql-parser",
-  "version": "3.0.2",
+  "version": "4.0.0-next.1",
```
Comment (jansule) [inline package.json:3]: should this be part of the `main` branch?

File: vitest.config.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+// import path from 'path';
```
Comment (jansule) [inline vitest.config.ts:1]: This can probably be removed

**Review by jansule — APPROVED** [review]
Very nice! Just a question and a minor remark

File: package.json (hunk 1 above)
Comment (hwbllmnn) [reply package.json:3]: No, this is currently published as a pre-release so I can use it for testing with the other repos.

File: vitest.config.ts (hunk 2 above)
Comment (hwbllmnn) [reply vitest.config.ts]: :heavy_check_mark:

**Comment by hwbllmnn** [comment]
Closing, replaced by #357

**Pull request closed without merging**